src/recording/macromanager.cpp

```diff
@@ -45,17 +47,30 @@ bool MacroManager::claimRecording() {
     return m_macroRecordingState.compare_exchange_weak(armed, MacroState::Recording);
 }
 
+void pollRecordingStart(MacroManager* pMacroManager) {
+    while (pMacroManager->getActiveChannel() == nullptr) {
+        QThread::msleep(300);
```

Comment:
Doesn't work. This will block a shared worker pool thread, which are also used for cover art loading.

Reply:
What should I do instead? I was also wondering what to use here, this is the only way I found.

Reply:
@daschuer any ideas?

Reply:
Use a timer

Reply:
To be clear, QTimer?

Reply:
Yes. And don't use a separate thread to access a QObject in a thread-unsafe way. Please get familiar with the Qt event loop.

Ideally a QObject is only accessed from its event loop thread. Same principles as for async executors apply, i.e. never block the event loop. Unfortunately we are violating these rules everywhere.

Reply:
Ah yes, QTimer is exactly what I wanted :)
see